resources/assets/js/components/Application/RedirectToLastTouchedStep.tsx

```diff
@@ -0,0 +1,119 @@
+import React from "react";
+import { useIntl } from "react-intl";
+import { useDispatch } from "react-redux";
+import { getLocale } from "../../helpers/localize";
+import {
+  useApplication,
+  useFetchAllApplicationData,
+  useJob,
+  useSteps,
+} from "../../hooks/applicationHooks";
+import { ApplicationStep } from "../../models/lookupConstants";
+import { loadingMessages } from "./applicationMessages";
+import BasicInfoPage from "./BasicInfo/BasicInfoPage";
+import ExperienceIntroPage from "./Experience/ExperienceIntroPage";
+import ExperiencePage from "./Experience/ExperiencePage";
+import FinalSubmitPage from "./FinalSubmit/FinalSubmitPage";
+import FitPage from "./Fit/FitPage";
+import IntroPage from "./Intro/IntroPage";
+import ReviewPage from "./Review/ReviewPage";
+import SkillsIntroPage from "./Skills/SkillsIntroPage";
+import SkillsPage from "./Skills/SkillsPage";
+
+interface RedirectToLastTouchedStepProps {
+  applicationId: number;
+  requestedStep?: ApplicationStep;
+  introStep?: boolean;
+}
+
+const RedirectToLastTouchedStep: React.FunctionComponent<RedirectToLastTouchedStepProps> = ({
```

Comment:
I have a few different thoughts about this component. 

My first thought was it would seem more appropriate to move this logic to the Laravel controller, do a redirect if necessary before the request is returned, and leave the ApplicationRoot router component as it was. This might be faster, since the redirect could be determined before the view is rendered.

If we do want to keep this Redirect component, then instead of just rendering the correct Step component, it should call the  redirect function in helpers/router.tsx, so that the url matches the step shown.

My last thought is that this all seems like overkill. I can understand not showing links in the Progress Bar to un-touched steps, but if someone goes through all the trouble of typing out the url, why stop them? We've built each step not to break if visited out of order.